drivers/hmis/app/graphql/mutations/create_draft_form.rb

```diff
@@ -0,0 +1,44 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  # This mutation is only for creating a new draft of a published form.
+  # To create a brand-new form, use CreateFormDefinition
+  class CreateDraftForm < CleanBaseMutation
+    argument :identifier, String, required: true
+
+    field :form_identifier, Types::Forms::FormIdentifier, null: true
+
+    def resolve(identifier:)
+      raise 'not allowed' unless current_user.can_manage_forms?
+
+      definitions = Hmis::Form::Definition.
+        exclude_definition_from_select.
+        where(identifier: identifier)
+
+      raise 'not found' if definitions.empty?
+
+      existing_drafts = definitions.where(status: Hmis::Form::Definition::DRAFT)
+      raise 'not allowed to create draft if one already exists' unless existing_drafts.empty?
+
+      # Re-fetch the most recent version (could be published or retired) to get the full form definition
+      last = Hmis::Form::Definition.find(definitions.max_by(&:version).id)
+
+      # Duplicate the last version, incrementing the version number and setting the status to draft
+      definition = last.dup
+      definition.version = last.version + 1
+      definition.status = Hmis::Form::Definition::DRAFT
+
+      if definition.valid?
+        definition.save!
+        { form_identifier: Hmis::Form::Definition.non_static.latest_versions.where(identifier: identifier).first }
+      else
+        errors.add_ar_errors(definition.errors)
+        { errors: errors }
+      end
```

Comment:
I don't think we want to return validation errors here. `definition.valid?` will just validate the status and identifier uniqueness, it does not validate the form structure itself. When we return `errors: [...]` they are meant to be validation errors that are correctible by the user. In this example, there are no errors that are fixable by the user. 

If there is an "unfixable" error, we prefer to raise that as an exception so that it gets sent to Sentry. `save!` will raise an exception if validation fails.

(We are not consistent about this, we often do what you propose here, so I see why you added it).


So my proposal would be to simplify to:

```suggestion
      definition.save!
      
      # return the form identifier, which will resolve the newly created draft as the draft_version
      { form_identifier: Hmis::Form::Definition.latest_versions.where(identifier: identifier).first }
```

Let me know if that makes sense / if you have questions

Reply:
Yes! I was sorta thinking something along these lines already (from my question [here](https://github.com/greenriver/hmis-frontend/pull/804/files#r1635505852) on the frontend PR) -- so it totally makes sense that the fix is actually here upstream.